presto-hive/src/test/java/com/facebook/presto/hive/HiveQueryRunner.java

```diff
@@ -78,19 +94,30 @@ public static DistributedQueryRunner createQueryRunner(Iterable<TpchTable<?>> ta
             metastore.createDatabase(createDatabaseMetastoreObject(baseDir, "tpch_sampled"));
 
             queryRunner.installPlugin(new HivePlugin(HIVE_CATALOG, metastore));
-            Map<String, String> hiveProperties = ImmutableMap.<String, String>builder()
+            ImmutableMap.Builder<String, String> hivePropertiesBuilder = ImmutableMap.<String, String>builder()
                     .put("hive.metastore.uri", "thrift://localhost:8080")
                     .put("hive.allow-add-column", "true")
                     .put("hive.allow-drop-table", "true")
                     .put("hive.allow-rename-table", "true")
                     .put("hive.allow-rename-column", "true")
                     .put("hive.time-zone", TIME_ZONE.getID())
-                    .put("hive.security", "sql-standard")
-                    .build();
-            queryRunner.createCatalog(HIVE_CATALOG, HIVE_CATALOG, hiveProperties);
-
-            copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, createSession(), tables);
-            copyTpchTables(queryRunner, "tpch_sampled", TINY_SCHEMA_NAME, createSampledSession(), tables);
+                    .put("hive.security", "sql-standard");
+            if (bucketed) {
+                hivePropertiesBuilder
+                        .put("hive.max-initial-split-size", "10kB")
+                        .put("hive.max-split-size", "10kB")
+                        .put("hive.storage-format", "TEXTFILE") // so that there's no minimum split size for the file
+                        .put("hive.compression-codec", "NONE"); // so that the file is splittable
+            }
+            queryRunner.createCatalog(HIVE_CATALOG, HIVE_CATALOG, hivePropertiesBuilder.build());
+
+            if (!bucketed) {
+                copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, createSession(), tables);
+                copyTpchTables(queryRunner, "tpch_sampled", TINY_SCHEMA_NAME, createSampledSession(), tables);
+            }
+            else {
+                copyTpchTablesBucketed(queryRunner, "tpch", TINY_SCHEMA_NAME, createSession(), tables);
```

Comment:
Instead of having a switch, can we just create bucketed tables in a new schema, and then tests can chose to use the bucketed tables explicitly?